### PR TITLE
Feature/direct and default pubsub

### PR DESCRIPTION
**Pull request by joniba**

### CODE REVIEW

File: src/queues/consumer.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            let consumer = consumers.pop();
+            return this.consume(consumer.handler, consumer.topic, consumer.options)
+        })
+    }
+
+    _validateConsumeChannel(queue) {
+        if (this.consumers.length) {
+            let sameQueueConsumer = this.consumers.find(x => x.queue === queue);
+            if (sameQueueConsumer)
+                throw new Error('Multiple consumers registered to the same queue. If you meant to add bindings, use the "channel.addTopics" method instead');
```
Comment (boaz-amit) [inline src/queues/consumer.js]: If I understand the error correctly, I think this message is more readable.
```suggestion
                throw new Error(`A consumer is already configured for queue "${queue}". If you meant to add bindings, use the "channel.addTopics" method instead`);
```

File: src/queues/publisher.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            let val = _.get(options, path);
+            if (typeof val === 'string')
+                _.set(options, path, ms(val));
+        });
+
+        return options;
+    }
+
+    _getAndVerifyDelayOptions(options) {
+        if (!_.isNumber(options.delay))
```
Comment (boaz-amit) [inline src/queues/publisher.js:84]: The `delay` cannot be an ms string? Not critical, but for the sake of consistency probably best to support later on.

File: test/common.js [hunk 3]
```diff
@@ -2,31 +2,53 @@ const Promise = require('bluebird');
...
+        console.log(`=============================== CLEANUP: deleting ${entities.length} ${entityType} ===============================`);
+
+        await Promise.each(entities, async entity => {
+            await this.deleteViaApi(entityType, entity.name, { addDefaultVhost });
         });
+    },
 
-        console.log(`=============================== CLEANUP: finished ===============================`);
+    async deleteViaApi(entityType, name, { addDefaultVhost = true } = {}) {
+        let vhost = addDefaultVhost ? '%2F/' : '';
```
Comment (boaz-amit) [inline test/common.js]: I still maintain that `%2F/` is unreadable to the average developer :)
```suggestion
        let vhost = addDefaultVhost ? encodeURIComponent('/')+'/' : '';
```

**Review by boaz-amit — APPROVED** [review]
@joniba Looks good. Thanks for this refactoring. Added three non-critical comments.

File: src/queues/publisher.js (hunk 2 above)
Comment (joniba) [reply src/queues/publisher.js:84]: awesome thank you!

**Pull request merged**